AppWidget/README.md

```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
 List of API changes
 ------------
 
-- **Widget description**
+# [Improve your app's widget picker experience](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#improve-widget-picker-experience)
 
-  In API level 31, a widget description is shown for each widget in the widget
-  picker if [`description`](app/src/main/res/xml/app_widget_info_grocery_list.xml\#L40) attribute is provided for your `appwidget-provider`
+## [Add scalable widget previews to the widget picker](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#add-scalable-widget-previews)
 
+  In Android 12, the widget preview displayed in the widget picker consists of a scalable preview, which you’ll provide as an XML layout set to the widget's default size. Previously, the widget preview was a static drawable resource, in some cases leading to previews not accurately reflecting widgets after they were added to the home screen.
+
+  To implement scalable widget previews, use the [previewLayout](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo?db=dac-team#previewLayout) attribute of the appwidget-provider element to provide an XML layout instead:
+
+  [`app_widget_info_checkbox_list`](app/src/main/res/xml/app_widget_info_checkbox_list.xml\#L50)
   ```xml
-  <appwidget-provider
-        android:description="@string/app_widget_grocery_list_description"
+    <appwidget-provider
+        android:previewLayout="@layout/widget_grocery_list"
     ... />
   ```
 
-- **Preview layout**
+## [Add a description for your widget](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#add-widget-description)
+
+  In Android 12, you can optionally provide a description for the widget picker to display for your widget. 
 
-  The widget preview is displayed from the layout XML if it's provided as
-  [`previewLayout`](app/src/main/res/xml/app_widget_info_grocery_list.xml\#L50) attribute for your `appwidget-provider` instead of a
-  static drawable resource using `previewImage` attribute.
-  Note that the preview may not reflect the widget after it's actually
-  placed (e.g. colors from themes)
+  Provide a description for your widget using the description attribute of appwidget-provider:
 
+  [`app_widget_info_checkbox_list`](app/src/main/res/xml/app_widget_info_checkbox_list.xml\#L40)
   ```xml
-    <appwidget-provider
-        android:previewLayout="@layout/widget_grocery_list"
+  <appwidget-provider
```

Comment:
Ditto, why indented?

Reply:
It wasn't intended. Removed the indentation.